src/data/upload.ts

```diff
@@ -143,25 +149,27 @@ function stringify(payload: EncodedPayload): string {
     return `{"e":${payload.e},"d":${payload.d}}`;
 }
 
-function send(data: string, sequence: number = null, last: boolean = false): void {
+function send(data: string, sequence: number, last: boolean): void {
     // Upload data if a valid URL is defined in the config
     if (config.url.length > 0) {
-        if (last && "sendBeacon" in navigator) {
+        // Upload data using sendBeacon only when it's the last payload and the data size is within safe limits.
+        // Most browsers in the wild will reject the sendBeacon call if data being transferred is over 64KB (BEACON_BYTES_LIMIT).
+        if (last && data.length < BEACON_BYTES_LIMIT && "sendBeacon" in navigator) {
```

Comment:
>BEACON_BYTES_LIMIT [](start = 34, length = 18)

Will the beacon call get an error or just be silently ignored?

Reply:
Silently ignored. In the network panel, browser doesn't even put this on the network and it keeps showing as (pending).

Reply:
Fixed as you suggested based on our offline conversation.